Counterexamples/CliffordAlgebra_not_injective.lean

```diff
@@ -0,0 +1,317 @@
+    add_zero]
+
+/-- `Q'`, lifted to operate on the quotient space `L`. -/
+@[simps!]
+def Q : QuadraticForm K L :=
+  QuadraticForm.ofPolar
+    (fun x =>
+      Quotient.liftOn' x Q' fun a b h => by
+        rw [Submodule.quotientRel_r_def] at h
+        suffices Q' (a - b) = 0 by rwa [Q'_sub, sub_eq_zero] at this
+        apply Q'_zero_under_ideal (a - b) h)
+    (fun a x => by
+      induction x using Quotient.inductionOn
+      exact Q'.toFun_smul a _)
+    (by rintro ⟨x⟩ ⟨x'⟩ ⟨y⟩; exact Q'.polar_add_left x x' y)
+    (by rintro c ⟨x⟩ ⟨y⟩; exact Q'.polar_smul_left c x y)
+
+open CliffordAlgebra
+
+/-- Basis vectors in the Clifford algebra -/
+def gen (i : Fin 3) : CliffordAlgebra Q := ι Q <| Submodule.Quotient.mk (Pi.single i 1)
+
+local notation "x'" => gen 0
+local notation "y'" => gen 1
+local notation "z'" => gen 2
+
+/-- The basis vectors square to one -/
+@[simp]
+theorem gen_mul_gen (i) : gen i * gen i = 1 := by
+  dsimp only [gen]
+  simp_rw [CliffordAlgebra.ι_sq_scalar, Q_apply, ← Submodule.Quotient.mk''_eq_mk,
+    Quotient.liftOn'_mk'', Q'_apply_single, mul_one, map_one]
+
+/-- By virtue of the quotient, terms of this form are zero -/
+theorem quot_obv : α • x' - β • y' - γ • z' = 0 := by
+  dsimp only [gen]
+  simp only [← (LinearMap.map_smul), ← (LinearMap.map_sub), ← Submodule.Quotient.mk_smul, ←
+    Submodule.Quotient.mk_sub]
+  convert LinearMap.map_zero _ using 2
+  rw [Submodule.Quotient.mk_eq_zero]
+  norm_num [sub_zero, Ideal.span, Pi.single_apply]
+
+set_option maxHeartbeats 400000 in
+/-- The core of the proof - scaling `1` by `α * β * γ` gives zero -/
+theorem αβγ_smul_eq_zero : (α * β * γ) • (1 : CliffordAlgebra Q) = 0 := by
+  suffices α • 1 - β • (y' * x') - γ • (z' * x') = 0 by
+    have := congr_arg (fun x => (β * γ) • x) this
+    dsimp only at this
+    simp_rw [smul_sub, smul_smul] at this
+    rwa [mul_assoc β γ γ, mul_right_comm β γ β, mul_right_comm β γ α, mul_comm β α, X_sq, X_sq,
+      zero_mul, mul_zero, zero_smul, zero_smul, sub_zero, sub_zero, smul_zero] at this
+  have : (α • x' - β • y' - γ • z') * x' = α • 1 - β • (y' * x') - γ • (z' * x') := by
+    simp_rw [sub_mul, smul_mul_assoc, gen_mul_gen]
+  rw [← this]
+  rw [quot_obv, MulZeroClass.zero_mul]
+
+theorem algebraMap_αβγ_eq_zero : algebraMap K (CliffordAlgebra Q) (α * β * γ) = 0 := by
+  rw [Algebra.algebraMap_eq_smul_one, αβγ_smul_eq_zero]
+
+/-- Our final result: for the quadratic form `Q60596.Q`, the algebra map to the clifford algebra
+is not injective, as it sends the non-zero `α * β * γ` to zero. -/
+theorem algebraMap_not_injective : ¬Function.Injective (algebraMap K <| CliffordAlgebra Q) :=
+  fun h => αβγ_ne_zero <| h <| by rw [algebraMap_αβγ_eq_zero, RingHom.map_zero]
+
+end Q60596
+
+open Q60596 in
+/-- The general statement: not every Clifford algebra over a module has an injective algebra map. -/
+theorem CliffordAlgebra.not_forall_algebraMap_injective.{v} :
+    -- TODO: make `R` universe polymorphic
```

Comment:
You could just make `M : Type` instead :-) (recall that this is the generality which mathematicians actually work in...)

Reply:
I thought it was a good API test to try and prove the universe-polymorphic version, so I'd rather keep `M : Type*`. At any rate, the interesting mathematical result is the `algebraMap_not_injective` a few lines above! This is just the "look, this statement is precisely the negation of a lemma you might hope to find in mathlib" as opposed to "determining that this constitutes a counterexample is left to the reader".